server/pulp/server/webservices/views/content.py

```diff
@@ -1,13 +1,366 @@
+        raise OperationPostponed(async_task)
+
+
+class DeleteOrphansActionView(View):
+    """
+    Deprecated in 2.4, please use the more restful OrphanResource delete instead
+
+    Delete all orphans.
+    """
+
+    @auth_required(authorization.DELETE)
+    def post(self, request):
+        """
+        Pass the appropriate tags and json object to delete_orphans_by_id.
+        Raises an OperationPostponed.
+
+        :param request: WSGI request object
+        :type  request: django.core.handlers.wsgi.WSGIRequest
+
+        :raises: OperationPostponed when an async operation is performed
+        """
+        all_orphans = request.body_as_json
+        task_tags = [tags.action_tag('delete_orphans'),
+                     tags.resource_tag(tags.RESOURCE_CONTENT_UNIT_TYPE, 'orphans')]
+        async_task = content_orphan.delete_orphans_by_id.apply_async([all_orphans], tags=task_tags)
+        raise OperationPostponed(async_task)
+
+
+class CatalogResourceView(View):
+    """
+    Views for the catalog by source_id.
+    """
+
+    @auth_required(authorization.DELETE)
+    def delete(self, request, source_id):
+        """
+        Delete all entries from the catlog that have the provided source id
+
+        :param request: WSGI request object
+        :type  request: django.core.handlers.wsgi.WSGIRequest
+        :param source_id: id of source whose content should be deleted
+        :type  source_id: str
+
+        :return: response containing a dict containing the number if items deleted
+        :rtype : django.http.HttpResponse
+        """
+        manager = factory.content_catalog_manager()
+        purged = manager.purge(source_id)
+        deleted_info = dict(deleted=purged)
+        return generate_json_response(deleted_info)
+
+
+class ContentTypeResourceView(View):
+    """
+    View for a single content type.
+    """
+
+    @auth_required(authorization.READ)
+    def get(self, request, type_id):
+        """
+        Return a response containing a dict with info about a content type.
+
+        :param request: WSGI request object
+        :type  request: django.core.handlers.wsgi.WSGIRequest
+        :param type_id: type of content unit
+        :type  type_id: str
+
+        :return: response containing a dict that contains info about a content type or
+                 404 response if the specified content type is not found.
+        :rtype : django.http.HttpResponse or HttpResponseNotFound
+        """
+        cqm = factory.content_query_manager()
+        content_type = cqm.get_content_type(type_id)
+        if content_type is None:
+            msg = _('No content type resource: %(r)s') % {'r': type_id}
+            return generate_json_response(msg, response_class=HttpResponseNotFound)
+        resource = serialization.content.content_type_obj(content_type)
+        # These links do not point to valid endpoints but are left here for for semantic versioning.
```

Comment:
Can a bug be filed on this problem? Then this code can reference the bug number.